pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -73,12 +73,20 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 	}
 
 	spp := dv.oc.Properties.ServicePrincipalProfile
-	token, err := aad.GetToken(ctx, dv.log, spp.ClientID, string(spp.ClientSecret), dv.subscriptionDoc.Subscription.Properties.TenantID, dv.env.Environment().ActiveDirectoryEndpoint, dv.env.Environment().ResourceManagerEndpoint)
+
+	spToken, err := aad.NewServicePrincipalToken(spp.ClientID, string(spp.ClientSecret), dv.subscriptionDoc.Subscription.Properties.TenantID, dv.env.Environment().ActiveDirectoryEndpoint, dv.env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return err
+	}
+
+	tokenAuthorizer := refreshable.NewAuthorizer(spToken)
+
+	err = aad.AuthenticateServicePrincipalToken(ctx, dv.log, tokenAuthorizer, nil)
```

Comment:
Since we always are creating a new SPtoken and then authenticating with a new Authorizer we could encapsulate this logic to enforce all three operations happen when we create a new token. This would DRY up `pkg/api/validate/openshiftcluster_validatedynamic.go`, pkg/api/validate/dynamic/serviceprincipal.go`, and `pkg/cluster/serviceprincipal.go`

Reply:
Hi @s-amann ,
Didn't change the existing logic in these area, its always creating a new token and authenticating it (you can check with master branch).
Do we really need to change that as part of this requirement and scope?

Reply:
Hi @s-amann ,
Did the changes, could you please review and let us know your comments?

Reply:
Not required, because that code is no longer changed in the PR with new changes